### PR TITLE
add appveyor

### PR DESCRIPTION
**Pull request by tylerl0706**
Adds CI for the PowerShell worker. It:

* does build
* invokes both xunit and pester tests
* uploads the nupkg to artifacts

### CODE REVIEW

**Comment by tylerl0706** [comment]
lights are green 🚥 full speed ahead!

File: build.ps1 [hunk 1]
```diff
@@ -58,31 +62,39 @@ if ($missingTools.Count -gt 0) {
...
-dotnet build -c $Configuration
-dotnet publish -c $Configuration
+    dotnet build -c $Configuration
+    dotnet publish -c $Configuration
 
-Push-Location package
-dotnet pack -c $Configuration
-Pop-Location
+    Push-Location package
+    dotnet pack -c $Configuration
```
Comment (daxian-dbw) [inline build.ps1]: Same here, get rid of `push/pop-location` by `dotnet pack $PSScriptRoot/package`.

File: build.ps1 [hunk 2]
```diff
@@ -58,31 +62,39 @@ if ($missingTools.Count -gt 0) {
...
+    Pop-Location
+}
 
 # Test step
-if($Test) {
+if($Test.IsPresent) {
     Push-Location test
     dotnet test
```
Comment (daxian-dbw) [inline build.ps1]: Same here, get rid of `push/pop-location` by `dotnet test $PSScriptRoot/test`

File: build.ps1 [hunk 3]
```diff
@@ -58,31 +62,39 @@ if ($missingTools.Count -gt 0) {
...
-    Invoke-Pester Modules
+
+    if($env:APPVEYOR) {
+        $res = Invoke-Pester Modules -OutputFormat NUnitXml -OutputFile TestsResults.xml -PassThru
+        (New-Object 'System.Net.WebClient').UploadFile("https://ci.appveyor.com/api/testresults/nunit/$($env:APPVEYOR_JOB_ID)", (Resolve-Path .\TestsResults.xml))
+        if ($res.FailedCount -gt 0) { throw "$($res.FailedCount) tests failed." }
+    } else {
+        Invoke-Pester Modules
```
Comment (daxian-dbw) [inline build.ps1]: same here, `Invoke-Pester $PSScriptRoot/test/Modules`

File: appveyor.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+image:
+- Ubuntu
+- Visual Studio 2017
+
+max_jobs: 1
+
+configuration: Debug
+
+install:
+  - pwsh: Install-Module PSDepend -Force -Scope CurrentUser
```
Comment (daxian-dbw) [inline appveyor.yml]: I don't see `dotnet SDK 2.1` installed. Is it pre-installed in the appveyor images?

Comment (tylerl0706) [reply appveyor.yml]: Yeah it comes by default.

File: build.ps1 (hunk 3 above)
Comment (tylerl0706) [reply build.ps1]: fixed

File: build.ps1 (hunk 2 above)
Comment (tylerl0706) [reply build.ps1]: fixed

File: build.ps1 (hunk 1 above)
Comment (tylerl0706) [reply build.ps1]: fixed

File: build.ps1 [hunk 5]
```diff
@@ -58,31 +62,39 @@ if ($missingTools.Count -gt 0) {
...
-if ((Get-Module -ListAvailable -Name PSDepend).Count -gt 0) {
-    Invoke-PSDepend -Path src -Force
-} else {
-    & "$PSScriptRoot/tools/InstallDependencies.ps1"
-}
-
-dotnet build -c $Configuration
-dotnet publish -c $Configuration
+    dotnet build -c $Configuration
+    dotnet publish -c $Configuration
```
[the remark this answers is not in the document]
Comment (tylerl0706) [reply build.ps1]: fixed

File: build.ps1 [hunk 6]
```diff
@@ -58,31 +62,39 @@ if ($missingTools.Count -gt 0) {
...
-# Install using PSDepend if it's available, otherwise use the backup script
-if ((Get-Module -ListAvailable -Name PSDepend).Count -gt 0) {
-    Invoke-PSDepend -Path src -Force
-} else {
-    & "$PSScriptRoot/tools/InstallDependencies.ps1"
-}
-
-dotnet build -c $Configuration
-dotnet publish -c $Configuration
+    dotnet build -c $Configuration
```
[the remark this answers is not in the document]
Comment (tylerl0706) [reply build.ps1]: fixed

**Comment by tylerl0706** [comment]
@daxian-dbw I removed a not-so-important test that was giving me trouble (because of the logger) that I also removed in the #51. Just a heads up.

File: build.ps1 [hunk 7]
```diff
@@ -54,37 +58,35 @@ if ($missingTools.Count -gt 0) {
...
 
-# Start at the root of the directory
-Push-Location $PSScriptRoot
-
 # Clean step
-if($Clean) {
+if($Clean.IsPresent) {
+    Push-Location $PSScriptRoot
     git clean -fdx
+    Pop-Location
```
[the remark this answers is not in the document]
Comment (tylerl0706) [reply build.ps1]: as discussed offline, we can leave it like it is because git doesn't let you `clean` outside of the git tree.

**Pull request merged**